src/hello_client.erl

```diff
@@ -165,12 +166,17 @@ handle_call(terminate, _From, State) ->
 %% @hidden
 handle_info({?INCOMING_MSG, Message}, State) ->
     incoming_message(Message, State);
+handle_info(?PING, State = #client_state{ping_counter = PingCounter}) when PingCounter >= 3 ->
+    ?LOG_ERROR("There is no PONG answer on PING for a long time. Client will be restarted.", []),
```

Comment:
@thz in this log message I wanted to be more clearer: we sent PONG 3 times and didn't receive any answers. For me,  `3 * keep_alive_interval` is a long time for one PONG answer. 
Of course, we could more specific and have timestamp for last PONG message (if there was).
And of course, we could get rid of PingCounter and PIngInterval combination and have something like PingNoAnswerInterval option, which parametrizes time-span in seconds to shutdown if there no PONG answer within. But, in my opinion, PingCounter is more OTP way (like supervisor restart politic)
I maybe don't understand you on 100%, can you give me more examples?